### DISCUSSION TITLE
I have a great idea for a chat app, all I need...

### DISCUSSION THREAD

**Discussion by mad-sol-dev** [FlowiseAI/Flowise#2237] [Ideas]
**Advanced Flowise Chat App:**

**a) Basic Chat Functions:**

The app should feature a chat system with a thread tree history where users can converse within created threads. Topic creators should have the ability to invite users to a topic and decide whether it should be private or visible to everyone in the thread history. Topics and their contained messages should be editable, and the creator should be able to selectively delete posts. The chat history should be searchable and sortable by date or topic (A-Z). It should also be possible to create sub-branches within a topic. Interactions should not only be manual but also possible via Whisper Speech-to-Text (STT) and Bark Text-to-Speech (TTS) functionalities (local provider or api). Additionally, file sharing between users should be available, and the chat should recognize common programming languages and text formats, displaying them appropriately in a code window or providing formatting tools or codes for such content.

**b) Integration with Telegram:**

The chat should also be accessible via a Telegram messenger client through a Telegram chat channel. Controlling the chat app through control characters should be possible, however, functionality should be limited; i.e., each user has a "Telegram" topic in their chat tree, may edit and delete content as usual, but is limited to a single chat without branches in this topic.

**c) Inclusion of Bots:**

It should be possible to invite bots to a topic, which are based on a Flowise flow. These bots would have access to the user's chat tree and the current chat history of the chat they have been invited to. Users should be able to choose whether the bot should react to their messages by selecting or deselecting a checkbox. Files sent by the user should be sent to the bot similarly to how they would be sent to another user, the user should be able to send voice messages and ask the bot to react to it via a voice / voice and text message.

----

Sadly GPT-Pilot only wastes a lot of money right now and I'm far away of being a good coder..

Kind of like in the Dilbert-Strip: "Boss: I have a great idea for a start up company. Boss: All I need is a seed investor and a engineer to do all the work. Employee: I think the economic term for what you have is 'nothing'."  :(

Does anyone know anything that comes close?

**Comment by automaton82** [comment]
Telegram has a bot API so it's probably possible, but the question is: why? What are you trying to accomplish?

By your description it sounds like you want a bot to sorta be "you" for awhile, answering chats so you don't have to. Or at least help answer. Is this correct?

**Reply by mad-sol-dev** [reply]
Hi auto,

so yes you can use bots / botfather quite easily, but - as far as I've read - you can't have secret chats with bots, only with clients. So if I have a local LLM, the chat system and a Telegram client, the conversations should stay pretty private. If it would be an option I would use Signal, but I've only seen a reverse engineered lib/client with features missing. However, it's not set that I won't add the feature to use Telegram bots (unless I give up, go bankrupt or crazy first ;) ).

**Reply by automaton82** [reply]
But if you want local chat, why even use Telegram? You can just run it on your machine and chat with it there without Telegram.

**Reply by mad-sol-dev** [reply]
Hey, no, with Flowise (sure indirectly) I can run LLMs locally and let them do whatever I want without having to pay for the API (but for the expensive hardware of course). By running it locally, I have full control, so I don't have to worry about an LLM in the field talking about my medical data at some point because it was trained with it. So how do I talk to my local model when I'm out on the road? How do I share it with my friends and family? I think the easiest way is to use a common messenger. (And for more advanced things there will be the chat system.)
